### DISCUSSION TITLE
LSP not getting attached on first install?

### DISCUSSION THREAD

**Discussion by ghost** [mason-org/mason.nvim#1255] [Q&A]
Hey!

I don't remember if this was normal behaviour but I'm using ```mason``` and ```mason-lspconfig``` and it seems that when I first install a LSP if won't get attached to the current buffer I'm in and I have to either do ```:e``` or close and open again. Just a minor annoyance but I'm wondering if that's the default or some other shenanigan having to do with my config.

I'm using ```lazy.nvim``` and my lsp spec looks like this:

```lua
return {
    "neovim/nvim-lspconfig",
    dependencies = {
        { "williamboman/mason.nvim", build = ":MasonUpdate" },
        "williamboman/mason-lspconfig.nvim",
        "hrsh7th/cmp-nvim-lsp",
        {
            "nvim-telescope/telescope.nvim",
            tag = "0.1.1",
            dependencies = { "nvim-lua/plenary.nvim" },
        },
        { "folke/neodev.nvim"},
    },
    config = function()
        -- key map shenanigans not included

        require("mason").setup()
        require("mason-lspconfig").setup()

        require("neodev").setup()

        local lspconfig = require("lspconfig")

        require("mason-lspconfig").setup_handlers({
            function(server_name)
                lspconfig[server_name].setup({
                    on_attach = lsp_attach,
                    capabilities = lsp_capabilities,
                })
            end,
        })
    end,
}

```

**Comment by RiteshChepuri** [comment]
You need to configure for each language on_attach = on_attach
here's what you need to do for all your lsp's
typescript.setup({
	server = {
		capabilities = capabilities,
		on_attach = on_attach,
	},
})

**Comment by williamboman** [comment]
> I don't remember if this was normal behaviour but I'm using `mason` and `mason-lspconfig` and it seems that when I first install a LSP if won't get attached to the current buffer I'm in and I have to either do `:e` or close and open again. Just a minor annoyance but I'm wondering if that's the default or some other shenanigan having to do with my config.

This is WAI. lspconfig uses the `FileType` autocmd in order to launch/attach servers, if mason-lspconfig were to automatically dispatch this autocmd after a server is installed it would potentially cause weird interoperability issues with other plugins. `:e` triggers `FileType`, `:doautoall FileType` should also do the trick.

You should be able to leverage Mason's event system to automatically do this:

```lua
local registry = require("mason-registry")

registry:on("package:install:success", function ()
  -- vim.schedule might not be needed 🤷‍♂️
  vim.schedule(function ()
    vim.cmd.doautoall { "FileType" }
  end)
end)
```
